Library/Formula/magit.rb

```diff
@@ -1,11 +1,24 @@
-require 'formula'
+class NewEnoughEmacs < Requirement
+  fatal true
+  default_formula "emacs"
 
-class Magit < Formula
-  homepage 'https://github.com/magit/magit'
-  url 'https://github.com/magit/magit/archive/1.2.1.tar.gz'
-  sha1 '3faeaab35934951a746e3be834d0457ca99bdc01'
+  # see https://github.com/magit/magit/tree/master#its-magit--a-git-porcelain-inside-emacs
+  # will require 24.4 upon release of magit 2.1.0
+  def satisfied?
+    major_version = `emacs --batch --eval "(princ emacs-major-version)"`
+    minor_version = `emacs --batch --eval "(princ emacs-minor-version)"`
+    (major_version + "." + minor_version).to_f >= 23.2
+  end
+
+  def message
+    "Emacs 23.2 or later is required for magit."
+  end
+end
```

Comment:
Currently only two other formulae have a custom emacs requirement—[`cask.rb`](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/cask.rb) and [`mu.rb`](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/mu.rb)—but there are others that could probably use it, like [`proof-general.rb`](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/proof-general.rb#L28).